pkg/cmd/server/api/secretstring.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	pemutil "github.com/openshift/origin/pkg/cmd/util/pem"
+)
+
+func GetExternalizableStringFileReferences(s *ExternalizableString) []*string {
+	if s == nil {
+		return nil
+	}
+	return []*string{
+		&s.File,
+		&s.KeyFile,
+	}
+}
+
+func GetExternalizableStringValue(s ExternalizableString) (string, error) {
```

Comment:
I'm open to better names for the type as well. `StringSource`? `ResolveStringSource()`?